astropy/modeling/functional_models.py

```diff
@@ -400,13 +400,15 @@ def evaluate(x, factor):
         return factor * x
 
 
-class Redshift(Fittable1DModel):
+class RedshiftWavelength(Fittable1DModel):
     """
     One dimensional redshift model.
 
+    Redshifts a wavelength array from rest (z=0) to the input redshift.
+
     Parameters
     ----------
-    z : float or a list of floats
+    z : int or float
```

Comment:
`model_set_axis` only applies for `n_models > 1`.  Admittedly this is confusing but it's two separate use cases.
In the future I might consider adding a separate interface for representing a collection of different models.